OpenSim/Simulation/Model/ElasticFoundationForce.cpp

```diff
@@ -1,385 +1,385 @@
-// * 
-// */
-//OpenSim::Array<std::string> ElasticFoundationForce::getRecordLabels() const 
-//{
-//    OpenSim::Array<std::string> labels("");
-//
-//    const ContactParametersSet& contactParametersSet = 
-//        get_contact_parameters();
-//
-//    for (int i = 0; i < contactParametersSet.getSize(); ++i)
-//    {
-//        ContactParameters& params = contactParametersSet.get(i);
-//        for (int j = 0; j < params.getGeometry().size(); ++j)
-//        {
-//            const ContactGeometry& geom =
-//                getModel().getComponent<ContactGeometry>(params.getGeometry()[j]);
-//            std::string frameName = geom.getFrame().getName();
-//            labels.append(getName()+"."+frameName+".force.X");
-//            labels.append(getName()+"."+frameName+".force.Y");
-//            labels.append(getName()+"."+frameName+".force.Z");
-//            labels.append(getName()+"."+frameName+".torque.X");
-//            labels.append(getName()+"."+frameName+".torque.Y");
-//            labels.append(getName()+"."+frameName+".torque.Z");
-//        }
-//    }
-//
-//    return labels;
-//}
-///*
-// * Provide the value(s) to be reported that correspond to the labels
-// */
-//OpenSim::Array<double> ElasticFoundationForce::getRecordValues(const SimTK::State& state) const 
-//{
-//    OpenSim::Array<double> values(1);
-//
-//    const ContactParametersSet& contactParametersSet = 
-//        get_contact_parameters();
-//
-//    const SimTK::ElasticFoundationForce& simtkForce = 
-//        (SimTK::ElasticFoundationForce &)(_model->getForceSubsystem().getForce(_index));
-//
-//    SimTK::Vector_<SimTK::SpatialVec> bodyForces(0);
-//    SimTK::Vector_<SimTK::Vec3> particleForces(0);
-//    SimTK::Vector mobilityForces(0);
-//
-//    //get the net force added to the system contributed by the Spring
-//    simtkForce.calcForceContribution(state, bodyForces, particleForces,
-//                                     mobilityForces);
-//
-//    for (int i = 0; i < contactParametersSet.getSize(); ++i)
-//    {
-//        ContactParameters& params = contactParametersSet.get(i);
-//        for (int j = 0; j < params.getGeometry().size(); ++j)
-//        {
-//            const ContactGeometry& geom =
-//                getModel().getComponent<ContactGeometry>(params.getGeometry()[j]);
-//    
-//            const auto& mbi = geom.getFrame().getMobilizedBodyIndex();
-//            const auto& thisBodyForce = bodyForces(mbi);
-//            SimTK::Vec3 forces = thisBodyForce[1];
-//            SimTK::Vec3 torques = thisBodyForce[0];
-//
-//            values.append(3, &forces[0]);
-//            values.append(3, &torques[0]);
-//        }
-//    }
-//
-//    return values;
-//}
-//
-//
-//} // end of namespace OpenSim
+/* -------------------------------------------------------------------------- *
+ *                    OpenSim:  ElasticFoundationForce.cpp                    *
```

Comment:
Do we support that with AD? I don't think so...I don't think we should uncomment if we do not support...